crates/bevy_ecs/src/change_detection.rs

```diff
@@ -428,6 +428,20 @@ impl<'a> MutUntyped<'a> {
     pub fn into_inner(self) -> PtrMut<'a> {
         self.value
     }
+
+    /// Returns a pointer to the value without taking ownership of this smart pointer or marking it as changed.
+    ///
+    /// In order to mark the value as changed, you need to call [`set_changed`](DetectChanges::set_changed) manually.
+    #[inline]
+    pub fn as_mut(&mut self) -> PtrMut<'_> {
```

Comment:
I don't like how this silently bypasses change detection. I see that `MutUntyped::into_inner` already does this, but I think we should fix that too.

For consistency with `Mut`, anything that might mutate the inner value should mark a change, and there should be an explicit `bypass_change_detection` method for getting around that.

Reply:
Also, this change should go in the migration guide.